File: Python/JumpSearch.py
import math

def jump_search(arr, target):
    n = len(arr)
    step = int(math.sqrt(n))
    prev, curr = 0, 0

    while curr < n and arr[curr] < target:
        prev = curr
        curr += step
        if curr >= n:
            break

    for i in range(prev, min(curr, n)):
        if arr[i] == target:
            return i
    return -1

File: JumpSearch.cpp
#include <iostream>
#include <cmath>
using namespace std;

int jumpSearch(int arr[], int n, int key) {
    int step = sqrt(n);
    int prev = 0;

    while (arr[min(step, n) - 1] < key) {
        prev = step;
        step += sqrt(n);
        if (prev >= n) return -1;
    }

    while (arr[prev] < key) {
        prev++;
        if (prev == min(step, n)) return -1;
    }

    return (arr[prev] == key) ? prev : -1;
}

int main() {
    int arr[] = {10, 20, 30, 40, 50};
    int n = sizeof(arr) / sizeof(arr[0]);
    int key = 30;

    int index = jumpSearch(arr, n, key);
    if (index != -1)
        cout << "Element found at index " << index;
    else
        cout << "Element not found";
}
